File: CarsMatter.Infrastructure/Helpers/BodyTypeHelper.cs
using CarsMatter.Infrastructure.Models.Enums;

namespace CarsMatter.Infrastructure.Helpers
{
    public class BodyTypeHelper
    {
        public static BodyType MapBodyType(string bodyTypeString)
        {
            switch (bodyTypeString)
            {
                case string bodyType when bodyTypeString.Contains("хэтчбек"):
                    return BodyType.Hatchback;
                case string bodyType when bodyTypeString.Contains("седан"):
                    return BodyType.Sedan;
                case string bodyType when bodyTypeString.Contains("универсал"):
                    return BodyType.Wagon;
                case string bodyType when bodyTypeString.Contains("люкс"):
                    return BodyType.Luxury;
                case string bodyType when bodyTypeString.Contains("кабриолет"):
                    return BodyType.Convertible;
                case string bodyType when bodyTypeString.Contains("купе"):
                    return BodyType.Coupe;
                case string bodyType when bodyTypeString.Contains("минивэн"):
                    return BodyType.Minivan;
                case string bodyType when bodyTypeString.Contains("внедорожник"):
                    return BodyType.SUV;
                default:
                    return BodyType.Sedan;
            }
        }
    }
}

File: CarsMatterApp/app/src/main/java/com/andrey/carsmatter/helpers/BodyTypeHelper.java
package com.andrey.carsmatter.helpers;

public class BodyTypeHelper {

    public static String MapBodyType(int bodyTypeNumber) {
        switch (bodyTypeNumber) {
            case 0:
                return "Хэтчбек";
            case 1:
                return "Седан";
            case 2:
                return "Универсал";
            case 3:
                return "Кабриолет";
            case 4:
                return "Купе";
            case 5:
                return "Минивэн";
            case 6:
                return "Люкс";
            case 7:
                return "AWD";
            case 8:
                return "Внедорожник";
            default:
                return "Неизвестный тип кузова";
        }
    }
}
